api/views.py

```diff
@@ -71,8 +74,19 @@ def post(self, request, *args, **kwargs):
         serializer = self.serializer_class(data=request.data, context={"request": request})
         serializer.is_valid(raise_exception=True)
         user = serializer.validated_data["user"]
+
         token, created = Token.objects.get_or_create(user=user)
-        return Response({"token": token.key, "user_id": user.pk, "email": user.email})
+
+        # If the existing token is older than AUTH_TOKEN_EXPIRATION, then recreate the object
+        token_expiry = token.created + timedelta(seconds=settings.AUTH_TOKEN_EXPIRATION)
+
+        if datetime.now(timezone.utc) > token_expiry:
+            print(f"INFO - Token expired as of {token_expiry}. Now generating a new token.")
+            token.delete()
+            token, created = Token.objects.get_or_create(user=user)
+            token_expiry = token.created + timedelta(seconds=settings.AUTH_TOKEN_EXPIRATION)
+
+        return Response({"token": token.key, "user_id": user.pk, "email": user.email, "expires": token_expiry})
```

Comment:
ultra minor thing. I guess it would be a bit faster to use the `created` variable to skip the expiry-checking if the token was just created. You could add a `if not created` statement around this, if you want to.